### PR TITLE
Implement `attachments.create`

### LINKED ISSUE #39: Implement `attachments.create` API
Reference: https://www.getoutline.com/developers#tag/Attachments/paths/~1attachments.create/post

This and the documents related endpoints are interesting, because that would allow us to import documents from other sources using this tool. The issue I had with attachments.create is that it actually does not upload the attachment - it just creates an entry in the database and returns needed credentials to upload the file to s3 our self.

### PR DESCRIPTION
**Pull request by zadjadr**
Closes #39

### CODE REVIEW

File: package_test.go [hunk 1]
```diff
@@ -235,9 +234,55 @@ func TestDocumentsClientCreate(t *testing.T) {
...
 	assert.Equal(t, &expected.Data, got)
-}  
-  
-  
+}
+
+func TestAttachmentsClientCreate(t *testing.T) {
+	testResponse := exampleAttachmentsCreateResponse
+
+	// Prepare HTTP client with mocked transport.
```
Comment (rsjethani) [inline package_test.go:242]: the `failed` test should also be there

**Review by rsjethani — CHANGES_REQUESTED** [review]

File: package_test.go (hunk 1 above)
Comment (zadjadr) [reply package_test.go:242]: You mean I should add a tests were we check a failed scenario?

Comment (StefMa) [reply package_test.go]: Added the test here: https://github.com/ioki-mobility/go-outline/pull/41/commits/03c7e922d72e10d915766971f41024bbc316d0c4

**Review by StefMa — APPROVED** [review]
Same as the [other PR](https://github.com/ioki-mobility/go-outline/pull/40#pullrequestreview-1586258874) 🙃 
I took over, did the requested changes forem you @rsjethani and updated the PR 👍 

In case it looks good, feel free to squash and merge 🚀 
For me its fine. Eventhough I have no idea how an attachment upload will look like (after we get the URL to upload).

**Comment by StefMa** [comment]
@rsjethani please have a look 😇

**Comment by zadjadr** [comment]
I asked already here: https://github.com/ioki-mobility/go-outline/issues/39 but maybe it makes sense to discuss it here?

If we "create" an attachment to a document, it will only create something in the database, but not actually upload the file to the storage backend. The user will get a response that can be used to upload to the backend. Do we want to keep it like that or also do the upload as well?

**Comment by StefMa** [comment]
Thanks for raising this up!

Our API should only reflect the real api from outline that's not. It
shouldn't be the responsibility of this project to directly upload the file.
However, we could implement this in the cli. So that the cli not only
creates the entry, but upload it afterwards as well...

**Comment by rsjethani** [comment]
yes agreed with @StefMa actual file upload should be a separate functionality

File: attachments.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	params := attachmentsCreateParams{Name: name, ContentType: contentType, Size: size}
+
+	return &AttachmentCreateClient{sl: copy, params: params}
+}
+
+// Create returns a client for creating a single attachment in the specified collection.
+// API reference: https://www.getoutline.com/developers#tag/Attachments/paths/~1attachments.create/post
+func (cl *AttachmentsClient) Create(name string, contentType string, size int) *AttachmentCreateClient {
+	return newAttachmentCreateClient(cl.sl, name, contentType, size)
+}
```
Comment (rsjethani) [inline attachments.go:46]: please move this below  newAttachmentsClient so as to maintain logical cohesion

**Review by rsjethani — CHANGES_REQUESTED** [review]

**Review by rsjethani — APPROVED** [review]
Thank you guys! LGTM

**Pull request merged**